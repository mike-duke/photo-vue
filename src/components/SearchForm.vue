<template>
  <section class="search-form">
    <h1 class="app-name">PhotoVue</h1>
    <h3>What would you like to see?</h3>
    <form @submit.prevent="handleSubmit">
      <input
        type="text"
        class="search-input"
        placeholder="Enter search term"
        v-model="searchTerm"
        autofocus>
      <button :disabled="!searchTerm.length">Submit</button>
    </form>
  </section>
</template>

<script>
export default {
  name: 'searchForm',
  data() {
    return {
      searchTerm: '',
    };
  },
  methods: {
    handleSubmit() {
      const lowerSearch = this.searchTerm.toLowerCase()
      this.$emit('addSearchTerm', lowerSearch);
      this.searchTerm = '';
    },
  },
};
</script>

<style lang="scss">
.search-form {
  background: rgba(20,66,112, 0.9);
  width: 50%;
  height: 50%;
  border-radius: 10px;
  box-shadow: 5px 5px 5px darkslategray;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 20px;
  text-align: center;

  .app-name {
    color: whitesmoke;
    margin: 10px 0;
    font-size: 3.5rem;
    text-shadow: 3px 2px 7px gray;
  }

  h3 {
    font-size: 1.2rem;
    color: whitesmoke;
    margin-bottom: 10px;
  }

  input {
    margin: 10px;
  }

  button {
    display: block;
    margin: 0 auto;
    background: darkslategray;
    color: whitesmoke;
  }

  button:disabled {
    background: gray;
  }

  @media screen and (max-width: 400px) {
    width: 75%;
  }
}
</style>
