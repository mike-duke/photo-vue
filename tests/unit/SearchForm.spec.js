import SearchForm from '@/components/SearchForm.vue';
import { mount } from '@vue/test-utils';

describe('SearchForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(SearchForm);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleSubmit', () => {
    it('should emit add-search-term with the correct args', () => {
      wrapper.setData({searchTerm: 'kitties'})
      wrapper.find('form').trigger('submit');
      expect(wrapper.emitted().addSearchTerm).toEqual([['kitties']])
    });

    it('should set searchTerm to an empty string', () => {
      expect(wrapper.vm.searchTerm).toEqual('');
      wrapper.setData({searchTerm: 'puppies'});
      expect(wrapper.vm.searchTerm).toEqual('puppies');
      wrapper.find('form').trigger('submit');
      expect(wrapper.vm.searchTerm).toEqual('')
    });
  });
});