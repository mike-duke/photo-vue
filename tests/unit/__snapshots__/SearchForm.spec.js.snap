// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<section class="search-form">
  <h1 class="app-name">PhotoVue</h1>
  <h3>What would you like to see?</h3>
  <form><input type="text" placeholder="Enter search term" autofocus="autofocus" class="search-input"> <button disabled="disabled">Submit</button></form>
</section>
`;
